master/internal/configpolicy/task_config_policy.go

```diff
@@ -24,3 +27,39 @@ type NTSCConfigPolicies struct {
 	InvariantConfig *model.CommandConfig `json:"invariant_config"`
 	Constraints     *model.Constraints   `json:"constraints"`
 }
+
+// PriorityAllowed returns true if the desired priority is within the task config policy limit.
+func PriorityAllowed(wkspID int, workloadType string, priority int, smallerHigher bool) (bool, error) {
+	// Check if a priority limit has been set in task config policies.
+	// Global policies have highest precedence.
+	limit, found, err := GetPriorityLimit(context.TODO(), nil, workloadType)
+	if err != nil {
+		// TODO do we really want to block on this?
+		return false, fmt.Errorf("unable to fetch task config policy priority limit")
+	}
+	if found {
+		return priorityWithinLimit(priority, limit, smallerHigher), nil
+	}
+
+	// TODO use COALESCE instead once postgres updates are complete.
```

Comment:
Should we ticket this TODO as a 1 pointer?

Reply:
We can, but it's a "nice-to-have" TODO. I left it so that if someone stumbles upon it in the future, they can optimize things ... but I don't think it's worth tracking it. WDYT?

Reply:
I agree, I don't really think this warrants a ticket. If someone wants later on, they can add this change. Thanks for commenting this potential optimization!

Reply:
cc @corban-beaird since you're working on the postgres updates! ^